subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/SearchUtil.java

```diff
@@ -37,23 +37,37 @@
      *         and only if the key is found.
      */
     public static <T> int binarySearch(List<T> sortedElements, SearchForComparator<T> searchForComparator) {
```

Comment:
This should be:

```java
public static <T> int binarySearch(List<T> sortedElements, Comparable<T> key) {
```

That would make the code read a lot easier. Also, you already refer to a "search key" in the Javadoc yet without it being present. Making this change introduces an actual key.

Reply:
Good point about using a `Comparable<T>` directly. For that to work, I needed to invert the results of the compare methods, since currently the search method uses `child.compareTo(key)` instead of `key.compareTo(child)`.

While at it, I removed `compareToPrefix`, since that comparison is not symmetric and the only usage we had can also be handled by `compareWithCommonPrefix`.